### PR TITLE
add LastUsedTime to the device list protocol

### PR DESCRIPTION
**Pull request by oconnor663**
r? @maxtaco

### CODE REVIEW

File: go/client/cmd_device_list.go [hunk 1]
```diff
@@ -52,10 +53,13 @@ func (c *CmdDeviceList) Run() error {
 
 func (c *CmdDeviceList) output(devs []keybase1.Device) {
 	w := GlobUI.DefaultTabWriter()
-	fmt.Fprintf(w, "Name\tType\tID\n")
-	fmt.Fprintf(w, "==========\t==========\t==========\n")
+	fmt.Fprintf(w, "Name\tType\tID\tCreated\tLast Used\n")
+	fmt.Fprintf(w, "==========\t==========\t==========\t==========\t==========\n")
```
Comment (oconnor663) [inline go/client/cmd_device_list.go]: This is getting a little wide, but not the end of the world.

```
$ keybase device list
Name                        Type         ID                                 Created               Last Used
==========                  ==========   ==========                         ==========            ==========
Paper Key (fruit roof...)   backup       82f0719f0446bf40ecb692c520ff0518   2016 Jun 1 11:24:42   2016 Jun 2 13:58:30
another threee              desktop      7c1fd4f0f02edd761235f7d970195818   2016 Jun 2 13:58:30   2016 Jun 2 13:58:35
home computer               desktop      c60911e1cb941529d1a50a7ceaa9a618   2016 Jun 1 11:24:42   2016 Jun 2 15:33:58
public name woo             desktop      f70f95a9096330f66f4dbe38bcac1618   2016 Jun 2 13:57:41   2016 Jun 2 13:57:45
```

File: go/libkb/sync_secret.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ type DeviceKey struct {
 	Status        int                  `json:"status"`
 	LksServerHalf string               `json:"lks_server_half"`
 	PPGen         PassphraseGeneration `json:"passphrase_generation"`
+	LastUsedTime  int64                `json:"last_used_time"`
```
Comment (oconnor663) [inline go/libkb/sync_secret.go]: This gets set in the unmarshal on line 138.

**Comment by oconnor663** [comment]
TODO: Invalidate the synced keys cache if the `last_used_time` field is missing.

**Comment by maxtaco** [comment]
👍

**Comment by oconnor663** [comment]
@maxtaco do you have time to eyeball that second commit?

File: go/libkb/sync_secret.go [hunk 3]
```diff
@@ -298,6 +302,19 @@ func (ss *SecretSyncer) DumpPrivateKeys() {
 	}
 }
 
+// As we add more fields to the data we're caching here, we need to detect the
+// cases where our cached data is missing the new fields. We can extend this
+// function with more cases as we add more fields.
+func (ss *SecretSyncer) cachedSyncedSecretsOutOfDate(cached *ServerPrivateKeys) bool {
+	for _, dev := range cached.Devices {
+		if dev.LastUsedTime == 0 {
+			G.Log.Debug("cachedSyncedSecretsOutOfDate noticed a cached device with no last used time")
```
Comment (maxtaco) [inline go/libkb/sync_secret.go]: please use local `G()`

**Comment by maxtaco** [comment]
One comment, please never use `G` again!

**Comment by maxtaco** [comment]
Otherwise 👍

File: go/libkb/sync_secret.go (hunk 3 above)
Comment (oconnor663) [inline go/libkb/sync_secret.go]: image

Comment (maxtaco) [inline go/libkb/sync_secret.go]: :)

**Pull request merged**